### ISSUE TITLE
Custom IPSW Loading

### ISSUE THREAD

**Issue by jeppojeps** [devos50/qemu#84]
I have unsuccessfully tried to get a modified ipsw or make my own, I thought since it uses OpeniBoot it should probably work...somehow.

**Comment by devos50** [comment]
@jeppojeps The IPSW is a zip file containing the files being used by the emulator (e.g., the root file system, iBoot, LLB etc). The emulator is currently not using the raw IPSW, so I'm not sure if making a custom IPSW would be helpful for the emulator.

**Comment by jeppojeps** [comment]
Yeah sure but a patcher for the ipsw includes patches for the pieces you
use to build and boot your version. They normally modified the entre
archive to make it easier for avg users. If I find one modified or I
achieve to do my custom, we could have cydia and other stuff for free. The
patched kernel really it should not make any difference but it should have
the 24K exploit for iBoot.

Cause otherwise I may try to inject just cydia in your current kernel since
sig checks are disabled and there are JB available. Though everything seems
too old.

**Comment by devos50** [comment]
I agree it's important to have a nice flow for the average user. For example, at some point I would really like some options where users can simply reboot/reflash/restore the entire device and provide their custom IPSWs. Unfortunately, the emulator currently is still very much a PoC with very limited functionality. I hope I'll have the opportunity to work towards this soon! 👍

**Comment by jeppojeps** [comment]
I went  pretty far on this with some old windows tools. But the firmware
version is really too old to support it. Though I was able to drag cydia
and appinstaller inside and half-launch them after patching AMFI. They
launched with a broken screen somehow, but the emulator did not crash.
